activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1105,6 +1105,51 @@ def uniq!(name)
       self
     end
 
+    # Excludes the specified record (or collection of records) from the resulting
+    # relation. For example:
+    #
+    #   Post.excluding(post)
+    #   # SELECT "posts".* FROM "posts" WHERE "posts"."id" != 1
+    #
+    #   Post.excluding(post_one, post_two)
+    #   # SELECT "posts".* FROM "posts" WHERE "posts"."id" NOT IN (1, 2)
+    #
+    # This can also be called on associations. As with the above example, either
+    # a single record of collection thereof may be specified:
+    #
+    #   post = Post.find(1)
+    #   comment = Comment.find(2)
+    #   post.comments.excluding(comment)
+    #   # SELECT "comments".* FROM "comments" WHERE "comments"."post_id" = 1 AND "comments"."id" != 2
+    #
+    # This is short-hand for <tt>.where.not(id: post.id)</tt> and <tt>.where.not(id: [post_one.id, post_two.id])</tt>.
+    #
+    # An <tt>ArgumentError</tt> will be raised if either no records are
+    # specified, or if any of the records in the collection (if a collection
+    # is passed in) are not instances of the same model that the relation is
+    # scoping.
+    def excluding(*records)
+      records.flatten!(1)
+
+      raise ArgumentError, "You must pass at least one #{klass.name} object to #excluding." if records.empty?
+
+      if records.any? { |record| !record.is_a?(klass) }
+        raise ArgumentError, "You must only pass a single or collection of #{klass.name} objects to #excluding."
+      end
+
+      spawn.excluding!(records)
+    end
+
+    def excluding!(records) # :nodoc:
+      # Treat single and multiple records differently in order to keep query
+      # clean in case of single record, ie, use != operator instead of NOT IN ().
+      if records.one?
+        where.not(primary_key => records.first.id)
+      else
+        where.not(primary_key => records)
+      end
```

Comment:
Not sure if we need this branch, or if there's a different way to express. I'll let @kamipo chime in, but that needn't block merging.

Reply:
The array predicate handler doesn't generate `IN` clause if the values has only one element. c6f08615154d4d89bca27d23573f293917b9f3c1